srv/modules/runners/cephprocesses.py

```diff
@@ -27,6 +27,10 @@ def check(cluster='ceph', **kwargs):
 
     roles = _cached_roles(search)
```

Comment:
It may be useful in this pull request to refactor this whole block from line 28-34 in the updated file.

Using the name `roles` to hold the cached roles and overwriting the variable (but not the cached roles) in the block below is confusing to read. For myself, as a late-comer, it is easy to misunderstand, and I could easily make a mistake understanding the code. Additionally, the test `if 'roles' in kwargs:` is actually unnecessary. Additionally, `k` is a non-descriptive variable name and should be changed.

This code also loops over kwargs['roles'] twice. These may as well be combined, although I don't think performance is a particular concern here.

```py
cached_roles = _cached_roles(search)
roles = {}
for role in kwargs.get('roles', {}):
    if role not in cached_roles:
        log.error('...')
        return False
    roles[role] = cached_roles[role]
```